File: AdventOfCode/2021/day1.py
# https://adventofcode.com/2021/day/1

with open('day1-input.txt') as f:
    data = [int(line.rstrip()) for line in f]
    # print(data)

increase_count = 0
prev = data[0]
for curr in data[1:]:
    if curr > prev:
        increase_count += 1
    prev = curr

print(f'Part 1: {increase_count}')

increase_count_triplet = 0
prev_triplet = sum(data[:3])
for i in range(4, len(data) + 1):
    curr_triplet = sum(data[i-3: i])
    if curr_triplet > prev_triplet:
        increase_count_triplet += 1
    prev_triplet = curr_triplet

print(f'Part 2: {increase_count_triplet}')


def part1():
    return sum(map(lambda x, y: x < y, data, data[1:]))


def part2():
    return sum(map(lambda x, y: x < y, data, data[3:]))


print(part1())
print(part2())

File: AdventOfCode/2022/aoc22/src/main/kotlin/day1.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun day1() {
    println("day 1")

//    val filename = "1-test.txt"
    val filename = "1-input.txt"
    val resourceStream = ClassLoader.getSystemResourceAsStream(filename)
    val reader = BufferedReader(InputStreamReader(resourceStream))

    // option 1
//    var line = reader.readLine()
//    println("$line")
//    while(line.isNotEmpty()){
//        line = reader.readLine()
//        println("$line")
//    }

    // option 2
//    val text = reader.readLines()

    val text = reader.readText()
    debugln(text)

    resourceStream.close()
    reader.close()

    var groups = text
        .split("\n\n")
    debugln(groups.toString())

    val intGroups = groups
        .map {
            debugln("groups $it")
            it.split("\n")
                .filter { it.isNotEmpty() }
                .map {
                    debugln("line $it")
                    it.toInt()
                }
        }
    debugln(intGroups.toString())

    val totals = intGroups.map { it.sum() }
    debugln(totals.toString())

    val part1 = totals.max()
    println("part 1 $part1")

    val part2 = totals.sortedDescending().subList(0, 3).sum()
    println("part 2 $part2")
}
